app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,7 @@
+class RecipesController < ApplicationController
+  def show
+    @recipe = Recipe.find(params[:id])
+    @ingredients = @recipe.ingredients
+    @total_cost = @ingredients.sum(:cost)
```

Comment:
You don't want this in here either. Your `.sum(:cost)` would live in a method called `total_cost` in your `recipe` model. Then you would call that method in your view on `@recipe`.